dist/images/README_IMAGE.md

```diff
@@ -0,0 +1,252 @@
+
+* The ovnkube binary is built in ../../go_controller and copied to this directory.
+
+* ovs/ovn evolves over time and a suitable version needs to be included in the image.
+  As of this writing ovs/ovn 2.11, or higher, is required. It is up to the image
+  builder to include a suitable version.
+
+* The base image is from any of many distributions. It is up to the image biulder to
+  select a suitable base. Depending on the base image, additional packages may be
+  needed to satisfy dependencies. There are sample Dockerfiles included here
+  that can be used as examples and to build development images. And a Makefile
+  to assist in the image build.
+
+There is some complexity in building the image. Each base may require additional
+packages to satisfy dependencies. In particular, a suitable version of ovs/ovn
+needs to be included. Also, automated build environments may impose
+additional requirements. The Dockerfiles in this repo will build images suitable for
+development and debug.
+
+The image must be stored in an image repository that can be accessed by pods in the
+cluster. The products that use ovn-kubernetes usually build the image and push it to
+a selected repository. The "image:" name in the daemonsets points to where the image
+was pushed.
+
+# Interface Versions
+
+Every container needs to set an environment variable that specifies the
+interface version that it is using.
+
+```
+env:
+- name: OVN_DAEMONSET_VERSION
+  value: "4"
+```
+At present version "4" and "3" are supported.
+
+# Version 4 interface
+
+## ovn-config configMap
+
+The ovn-config configMap is mounted into the container in directory
+```
+/var/run/ovn-config
+```
+The following files contain the config information:
+```
+k8s_apiserver
+net_cidr
+svc_cidr
+```
+In the yaml files each container has the volumeMounts: and the pod volumes: has configMap:
+```
+containers:
+- name:
+  volumeMounts:
+  - mountPath: /var/run/ovn-config
+    name: ovn-config-mount
+volumes:
+- configMap:
+    defaultMode: 420
+    items:
+    - key: k8s_apiserver
+      path: k8s_apiserver
+    - key: net_cidr
+      path: net_cidr
+    - key: svc_cidr
+      path: svc_cidr
+    name: ovn-config
+  name: ovn-config-mount
+```
+The ovn-config configMap is as follows:
+```
+apiVersion: v1
+data:
+  k8s_apiserver: https://wsfd-netdev22.ntdv.lab.eng.bos.redhat.com:8443
+  net_cidr: 10.2.0.0/12
+  svc_cidr: 10.3.0.0/20
+kind: ConfigMap
+metadata:
+  name: ovn-config
+  namespace: ovn-kubernetes
+```
+
+## ovnkube-config configMap
+
+The ovnkube-config configMap is mounted into the container in directory
+```
+/var/run/ovnkube-config
+```
+In the yaml files each container has the volumeMounts: and the pod volumes: has configMap:
+```
+containers:
+- name:
+  volumeMounts:
+  - mountPath: /var/run/ovnkube-config
+    name: ovnkube-config-mount
+volumes:
+- configMap:
+    defaultMode: 420
+    items:
+    - key: config
+      path: ovn_k8s.conf
```

Comment:
If you either rename the key to `ovn_k8s.conf` in the ConfigMap, or else change `ovnkube.sh` to expect `/var/run/ovnkube-config/config` instead of `.../ovn_k8s.conf`, then you wouldn't need the `items` section here. (Likewise, if you keep the `ovn-config` configmap, you don't need the `items` section when mounting that at all, since you're mounting every item, and not renaming any of them.)